### PR TITLE
Project view was added, spaghetti code refactor

### PR DESCRIPTION
**Pull request by cristian-bts**

### CODE REVIEW

File: BTSPocket/Controllers/Login/LoginViewController.swift [hunk 1]
```diff
@@ -23,11 +23,16 @@ class LoginViewController: UIViewController {
...
     private var authorizationError: NSError?
+    var expiredToken: Bool = false
     
     // MARK: LYFE CYCLE
     override func viewDidLoad() {
         super.viewDidLoad()
         self.setup()
+        
+        if expiredToken {
+            MessageManager.shared.showBar(title: "Expired session", subtitle: "Your session has expired", type: .info, containsIcon: true, fromBottom: false)
```
Comment (manuelsalinas-legion) [inline BTSPocket/Controllers/Login/LoginViewController.swift]: Add in message
**Your session has expired. Please, login again.**

File: BTSPocket/Controllers/Project/ProjectsViewController.swift [hunk 2]
```diff
@@ -8,21 +8,166 @@
 import UIKit
 
 class ProjectsViewController: UIViewController {
-
+    
+    // MARK:- Outlets and Variables
+    @IBOutlet weak var tableViewProjects: UITableView!
```
Comment (manuelsalinas-legion) [inline BTSPocket/Controllers/Project/ProjectsViewController.swift]: missing private if needed

File: BTSPocket/Controllers/Project/TableViewCells/ProjectTableViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+//  Created by bts on 20/05/21.
+//
+
+import UIKit
+import Kingfisher
+
+class ProjectTableViewCell: UITableViewCell {
+
+    // MARK: Outlets and variables
+    @IBOutlet weak var imageViewProject: UIImageView!
```
Comment (manuelsalinas-legion) [inline BTSPocket/Controllers/Project/TableViewCells/ProjectTableViewCell.swift]: missing private access control for outlets

File: BTSPocket/Controllers/Project/TableViewCells/ProjectTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    func setProjectValues(_ project: Project?) {
+        var countUsersMessage: String?
+        let countUsers = (project?.users?.count ?? 1) - 1
+        if countUsers == 1 {
+            countUsersMessage = "Only You"
+        } else {
+            countUsersMessage = "\(countUsers) users and you"
+        }
+        self.labelNameProject.text = project?.name
+        self.labelCountUsers.text = countUsersMessage
```
Comment (manuelsalinas-legion) [inline BTSPocket/Controllers/Project/TableViewCells/ProjectTableViewCell.swift]: missing format style via attributedText

**N users** and **you**

File: BTSPocket/Controllers/Team/TeamViewController.swift [hunk 5]
```diff
@@ -7,32 +7,61 @@
...
 class TeamViewController: UIViewController {
     
     // MARK:- Outlets and Variables
     @IBOutlet weak private var tableViewTeam: UITableView!
-//    @IBOutlet weak var searchBar: UISearchBar!
     
-    private var currentPage: Int = 0
-    private var totalPages: Int?
+    private var searchbarController = UISearchController()
+    private var refreshControl = UIRefreshControl()
```
Comment (manuelsalinas-legion) [inline BTSPocket/Controllers/Team/TeamViewController.swift:26]: Good. did you add a refresh control and behavior on profile?

File: BTSPocket/Controllers/Team/TeamViewController.swift [hunk 6]
```diff
@@ -53,44 +87,96 @@ class TeamViewController: UIViewController {
...
+        case .projectUsers:
+            return self.projectUsers?.count ?? 0
+        }
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withClass: UserTableViewCell.self)
-        let user = self.allUsers?[indexPath.row]
-        cell.setUserInfo(user)
+        // hacer ternacio a base de mode y eviar a la celda el tipo de usuario que necesita
```
Comment (manuelsalinas-legion) [inline BTSPocket/Controllers/Team/TeamViewController.swift]: missing Ternary  way

**Review by manuelsalinas-legion — CHANGES_REQUESTED** [review]
In general good work. You have a good progress now.
We need to discuss some pending tasks

**Comment by manuelsalinas-legion** [comment]
**ISSUES DETECTED ON UI**
@cristian-bts 

iPhone_12

File: BTSPocket/Controllers/Project/ProjectsViewController.swift [hunk 7]
```diff
@@ -8,21 +8,166 @@
...
+        vcTeam.navigationItem.searchController = nil
+        self.navigationController?.pushViewController(vcTeam, animated: true)
+    }
+}
 
+extension ProjectsViewController: UISearchBarDelegate {
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        if let text = searchBar.text {
+            if !text.isEmpty {
+                if text.count >= 3 {
```
Comment (manuelsalinas-legion) [inline BTSPocket/Controllers/Project/ProjectsViewController.swift]: Make value 3 into a constant for easy management. 
Sample: 

`let kMinimumCharactersForSearch: Int = 3`

File: BTSPocket/Controllers/Project/ProjectsViewController.swift [hunk 8]
```diff
@@ -8,21 +8,166 @@
...
+            } else {
+                self.getProjectsByUser(.firstPage)
+            }
+        }
+    }
+    
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        if let text = searchBar.text?.trim() {
+            if !text.isEmpty {
+                if text.count < 3 {
```
Comment (manuelsalinas-legion) [inline BTSPocket/Controllers/Project/ProjectsViewController.swift]: Make value 3 into a constant for easy management. 
Sample: 

`let kMinimumCharactersForSearch: Int = 3`

File: BTSPocket/Controllers/Project/TableViewCells/ProjectTableViewCell.swift [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+        let countUsers = (project?.users?.count ?? 1) - 1
+        if countUsers == 1 {
+            countUsersMessage = "Only You"
+        } else {
+            countUsersMessage = "\(countUsers) users and you"
+        }
+        let attributedMessage = NSMutableAttributedString(string: countUsersMessage!)
+        attributedMessage.setColor(countUsersMessage!, color: .btsBlue())
+        
+        self.labelNameProject.text = project?.name
```
Comment (manuelsalinas-legion) [inline BTSPocket/Controllers/Project/TableViewCells/ProjectTableViewCell.swift]: capitalized style

File: BTSPocket/Controllers/Project/TableViewCells/ProjectTableViewCell.swift [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+            let url = URL(string: Constants.urlBucketImages + photoPath)
+            // adding header in request.
+            let modifier = AnyModifier { request in
+                var mutableRequest = request
+                mutableRequest.setValue(Constants.serverAddress, forHTTPHeaderField: "Referer")
+                return mutableRequest
+            }
+            self.imageViewProject.kf.indicatorType = .activity
+            self.imageViewProject.kf.setImage(with: url,
+                                           placeholder: UIImage(named: "placeholderUser"),
```
Comment (manuelsalinas-legion) [inline BTSPocket/Controllers/Project/TableViewCells/ProjectTableViewCell.swift]: I will provide you a placeholder for projects if picture for project is null

File: BTSPocket/Controllers/Team/TeamViewController.swift [hunk 11]
```diff
@@ -7,32 +7,61 @@
 
 import UIKit
 
+enum ReloadType {
```
Comment (manuelsalinas-legion) [inline BTSPocket/Controllers/Team/TeamViewController.swift]: Public or private enums?

File: BTSPocket/Controllers/Team/TeamViewController.swift [hunk 12]
```diff
@@ -7,32 +7,61 @@
...
+    case firstPage
+    case refresh(Int, String)
+    case nextPage(Int, String)
+}
+
+enum TeamListStyle {
```
Comment (manuelsalinas-legion) [inline BTSPocket/Controllers/Team/TeamViewController.swift]: Public or private enums?

**Review by manuelsalinas-legion — CHANGES_REQUESTED** [review]

File: BTSPocket/Controllers/Team/TeamViewController.swift (hunk 11 above)
Comment (cristian-bts) [reply BTSPocket/Controllers/Team/TeamViewController.swift]: public

File: BTSPocket/Controllers/Team/TeamViewController.swift (hunk 12 above)
Comment (cristian-bts) [reply BTSPocket/Controllers/Team/TeamViewController.swift]: public

**Review by manuelsalinas-legion — APPROVED** [review]

**Pull request merged**